### PR TITLE
CRM-18465: fully resolve contributions before resolving tokens

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
- [CRM-18465: Contribution summary letters - fully resolve contributions before calling hook](https://issues.civicrm.org/jira/browse/CRM-18465)

### CODE REVIEW

File: CRM/Contribute/Form/Task/PDFLetterCommon.php [hunk 1]
```diff
@@ -265,6 +258,22 @@ public static function buildContributionArray($groupBy, $form, $returnProperties
...
+    foreach ($contacts as $contactID => $contact) {
+      $tokenResolvedContacts = CRM_Utils_Token::getTokenDetails(array('contact_id' => $contactID),
+        $returnProperties,
+        $skipOnHold,
+        $skipDeceased,
+        NULL,
+        $messageToken,
+        $task
+      );
+      $contacts[$contactID] = array_merge($tokenResolvedContacts[0][$contactID], $contact);
```
Comment (monishdeb) [inline CRM/Contribute/Form/Task/PDFLetterCommon.php]: I think that would be $tokenResolvedContacts[$contactID] as per the earlier change

Comment (eileenmcnaughton) [inline CRM/Contribute/Form/Task/PDFLetterCommon.php]: @monishdeb I checked & the  $tokenResolvedContacts does have that extra layer for some reason. I added a unit test to demonstrate it working (& lock it in)

**Comment by monishdeb** [comment]
Cool ... nice work :)

**Comment by monishdeb** [comment]
Tested works fine

**Pull request merged**